onl: clean up build and reduce amount of variables needed

KanjiMonster opened the pull request:
Make use of the ONL build system to autodetect build paths, which allows
us to reduce the amount of variables significantly, basically reducing the
need to define any ONL_* variables by machines.

For each platform in ONL there exists a package named onlp-<platform>.
Using the ONL package manager (onlpm), we can find out where in the
filesystem this package is located, and do not need to hardcode the path
to it anymore.

Sideeffect is a slight increase in build time, as it needs to build its
package database on the first invocation, which delays the build by a
few seconds.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

rubensfig commented:
Build fails with the following error on the 4630 platform.

```
make: Leaving directory '/tmp/accton-as4630-54pe/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/git/packages/base/any/onlp/builds/onlpd'
Traceback (most recent call last):
File "tools/onlpm.py", line 23, in <module>
       import lsb_release
       ImportError: No module named lsb_release
WARNING: exit code 1 from a shell command.
ERROR: Execution of '/tmp/accton-as4630-54pe/work/accton_as4630_54pe-poky-linux/onl/1.1+gitAUTOINC+dd61cc1792_8621a0bab7_4bd372b724-r0/temp/run.do_compile.23331' failed with exit code 1
```

KanjiMonster commented:
This python2 thing is getting annoying. Let's fix it. Will rebase onto #52, then this won't be an issue anymore.

KanjiMonster commented:
Updated and rebased on master with the switch to python3:

* fixed some leftovers in the python3 conversion patch
* removed the need for the lsb_release python module
* dropped BISDN_ONIE_PLATFORM

KanjiMonster commented on recipes-extended/onl/files/onl/0008-onlpm-add-an-argument-for-just-printing-the-builddir.patch:
```diff
@@ -0,0 +1,49 @@
+Subject: [PATCH 2/3] onlpm: add an argument for just printing the builddirs
+
+To allow building without packaging, we need to know the appropriate
+build dir execute make in.
```
Reworded slightly.

ideaship reviewed (approved):
LGTM
